### PR TITLE
BE : LOK-2545 : Follow Up to LOK-2503 task, requires UI requirement

### PR DESCRIPTION
**Pull request by Junaid-Khan-Nant**
## Description
Any is not supported in a GraphQL response. You have multiple routes to choose from here:

Serialize the config to JSON and make the field a string
Go all in and figure out how to represent arbitrary scalar data in GraphQL (maybe a JsonNode could work)
Create a follow up issue which we can defere until we know what UI needs there

## Jira link(s)
https://opennms.atlassian.net/browse/LOK-2545

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: rest-server/src/main/java/org/opennms/horizon/server/model/inventory/MonitoredEntityDTO.java [hunk 1]
```diff
@@ -30,5 +30,4 @@ public class MonitoredEntityDTO {
     public String entityId;
     public long locationId;
     public String type;
-    // public Any config; // TODO : need to be change LOK-2545
```
Comment (fooker) [inline rest-server/src/main/java/org/opennms/horizon/server/model/inventory/MonitoredEntityDTO.java:33]: We still need this to be added. That was the whole point of the issue after all.

File: shared-lib/inventory/src/main/proto/monitoredEntitySearch.proto [hunk 2]
```diff
@@ -40,7 +40,7 @@ message MonitorEntityResponse {
   string entityId = 1;
   int64 locationId = 2;
   string type = 3;
-  //google.protobuf.Any config = 4; // TODO: need to be change LOK-2545
+  string config = 4;
```
Comment (fooker) [inline shared-lib/inventory/src/main/proto/monitoredEntitySearch.proto:43]: I don't see why we need to change the proto here. Using `Any` in protobuf is perfectly fine.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/monitoring/MonitoredEntityService.java [hunk 3]
```diff
@@ -67,10 +70,21 @@ public void publishTaskSet(final String tenantId, final long locationId) {
                         .setSchedule(TaskUtils.DEFAULT_SCHEDULE)
                         .setMonitoredEntityId(entity.getId())
                         .setPluginName(entity.getType() + "Monitor") // TODO fooker: Make sure this matches
-                        .setConfiguration(entity.getConfig())
+                        .setConfiguration(getMonitorConfiguration(entity.getType(), entity.getConfig()))
```
Comment (fooker) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/monitoring/MonitoredEntityService.java:73]: This is confusing. I don't understand why the roundtrip for using JSON in the monitored entity API is required.
From my understanding the transformation from `Any` to JSON is only required in the GraphQL interface.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/monitoring/MonitoredEntityService.java [hunk 4]
```diff
@@ -67,10 +70,21 @@ public void publishTaskSet(final String tenantId, final long locationId) {
...
                         .build())
                 .toList();
 
         this.taskSetPublisher.publishNewTasks(tenantId, locationId, tasts);
     }
+
+    private Any getMonitorConfiguration(String monitorType, String jsonConfig) {
+        switch (monitorType) {
```
Comment (fooker) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/monitoring/MonitoredEntityService.java:81]: The generic implementation must not depend on the monitor type. But I think this is not required anyways.

**Review by fooker — CHANGES_REQUESTED** [review]
IMO, this does not solve the Issue we have here. There is no need to change the type of the proto send to minion but we need to serialize for the GraphQL interfaces.

**Review by fooker — CHANGES_REQUESTED** [review]
Please remove the commented out line. I still would like to see this not transported as a string. Maybe @scthompsonopen has a clue how this works.

File: inventory/main/src/main/java/org/opennms/horizon/inventory/mapper/MonitorEntityMapper.java [hunk 5]
```diff
@@ -21,14 +21,38 @@
...
 import org.opennms.horizon.inventory.dto.MonitorEntityResponse;
 import org.opennms.horizon.inventory.monitoring.MonitoredEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Mapper(componentModel = "spring")
 public interface MonitorEntityMapper extends DateTimeMapper {
 
-    MonitoredEntity dtoToModel(MonitorEntityResponse dto);
+    Logger LOGGER = LoggerFactory.getLogger(MonitorEntityMapper.class);
```
Comment (fooker) [inline inventory/main/src/main/java/org/opennms/horizon/inventory/mapper/MonitorEntityMapper.java:38]: LOGGER is not used.

**Comment by Junaid-Khan-Nant** [comment]
> Please remove the commented out line. I still would like to see this not transported as a string. Maybe @scthompsonopen has a clue how this works.

With response type as a JsonObject :
Screenshot from 2024-06-12 17-13-02
With response type as a string :
str

File: shared-lib/inventory/src/main/proto/monitoredEntitySearch.proto [hunk 6]
```diff
@@ -40,7 +40,7 @@ message MonitorEntityResponse {
   string entityId = 1;
   int64 locationId = 2;
   string type = 3;
-  //google.protobuf.Any config = 4; // TODO: need to be change LOK-2545
+  google.protobuf.Struct config = 4;
```
Comment (fooker) [inline shared-lib/inventory/src/main/proto/monitoredEntitySearch.proto:43]: By comments to remove, I mean this.

File: rest-server/src/main/java/org/opennms/horizon/server/model/inventory/MonitoredEntityDTO.java [hunk 7]
```diff
@@ -30,5 +30,5 @@ public class MonitoredEntityDTO {
...
+    public String config;
```
Comment (fooker) [inline rest-server/src/main/java/org/opennms/horizon/server/model/inventory/MonitoredEntityDTO.java:33]: By comments to remove, I mean this.

**Review by fooker — CHANGES_REQUESTED** [review]

**Comment by fooker** [comment]
@Junaid-Khan-Nant I will convert this to a draft and move this to backlog until we have clear requirements.

**Pull request closed without merging**